### PR TITLE
add format test for if/for statements

### PR DESCRIPTION
**Pull request by liangdzou**
### Your checklist for this pull request

Please review the [guidelines for contributing](http://solidity.readthedocs.io/en/latest/contributing.html) to this repository.

Please also note that this project is released with a [Contributor Code of Conduct](CONDUCT.md). By participating in this project you agree to abide by its terms.

### Checklist
- [x] Code compiles correctly
- [ ] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### Description

add format checkers

### CODE REVIEW

File: scripts/check_if_for_format.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+(
+cd $REPO_ROOT
+FORMATERROR=$(git grep -nIE "\<if\(" -- '*.h' '*.cpp')
+FORMATERROR=$FORMATERROR"\n"$(git grep -nIE "\<for\(" -- '*.h' '*.cpp')
+FORMATERROR=$FORMATERROR"\n"$(git grep -nIE "\<if\>\s*\(.*\)\s*\{\s*$" -- '*.h' '*.cpp')
+FORMATERROR=$FORMATERROR"\n"$(git grep -nIE "\<for\>\s*\(.*\)\s*\{\s*$" -- '*.h' '*.cpp')
+
+FORMATERROR=$(echo -e "$FORMATERROR" | grep -v "^[a-zA-Z\./]*:[0-9]*:\s*\/\/")
+FORMATERROR=$(echo -e "$FORMATERROR" | grep -v "^[a-zA-Z\./]*:[0-9]*:\s*\/\*")
+FORMATERROR=$(echo -e "$FORMATERROR" | grep -v "libdevcore")
```
Comment (chriseth) [inline scripts/check_if_for_format.sh]: Why do you exclude this?

File: scripts/check_if_for_format.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+REPO_ROOT="$(dirname "$0")"/..
+
+(
+cd $REPO_ROOT
+FORMATERROR=$(git grep -nIE "\<if\(" -- '*.h' '*.cpp')
+FORMATERROR=$FORMATERROR"\n"$(git grep -nIE "\<for\(" -- '*.h' '*.cpp')
+FORMATERROR=$FORMATERROR"\n"$(git grep -nIE "\<if\>\s*\(.*\)\s*\{\s*$" -- '*.h' '*.cpp')
+FORMATERROR=$FORMATERROR"\n"$(git grep -nIE "\<for\>\s*\(.*\)\s*\{\s*$" -- '*.h' '*.cpp')
+
+FORMATERROR=$(echo -e "$FORMATERROR" | grep -v "^[a-zA-Z\./]*:[0-9]*:\s*\/\/")
```
Comment (chriseth) [inline scripts/check_if_for_format.sh]: Which files does this match?

File: scripts/check_if_for_format.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+REPO_ROOT="$(dirname "$0")"/..
+
+(
+cd $REPO_ROOT
+FORMATERROR=$(git grep -nIE "\<if\(" -- '*.h' '*.cpp')
```
Comment (chriseth) [inline scripts/check_if_for_format.sh]: Have you tried something like this:
```
FORMATERROR=$(
(
git grep -nIE "\<if\(" -- '*.h' '*.cpp'
git grep -nIE "\<for\(" -- '*.h' '*.cpp'
git grep -nIE "\<if\>\s*\(.*\)\s*\{\s*$" -- '*.h' '*.cpp'
git grep -nIE "\<for\>\s*\(.*\)\s*\{\s*$" -- '*.h' '*.cpp'
) | grep -v "^[a-zA-Z\./]*:[0-9]*:\s*\/\/" | grep -v "^[a-zA-Z\./]*:[0-9]*:\s*\/\*" | grep -v "^test/contracts/"
)
```

File: scripts/check_if_for_format.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+FORMATERROR=$FORMATERROR"\n"$(git grep -nIE "\<for\(" -- '*.h' '*.cpp')
+FORMATERROR=$FORMATERROR"\n"$(git grep -nIE "\<if\>\s*\(.*\)\s*\{\s*$" -- '*.h' '*.cpp')
```
Comment (chriseth) [inline scripts/check_if_for_format.sh]: What about combining these to `(if|for)`?

**Review by chriseth — CHANGES_REQUESTED** [review]
Please combine these tests with the checks for trailing whitespace and name the file "check_style.sh" or something like that.

File: scripts/check_if_for_format.sh (hunk 1 above)
Comment (liangdzou) [reply scripts/check_if_for_format.sh]: the first two are comments, I think it is reasonable to ignore comments

shall we modify the code in libdevcore?
will we need to sync with upstream somehow?

File: scripts/check_if_for_format.sh (hunk 2 above)
Comment (liangdzou) [reply scripts/check_if_for_format.sh]: line 12 matches comments // ...
line 13 is /*

File: scripts/check_if_for_format.sh (hunk 3 above)
Comment (liangdzou) [reply scripts/check_if_for_format.sh]: looks much better 👍

File: scripts/check_if_for_format.sh (hunk 4 above)
Comment (liangdzou) [reply scripts/check_if_for_format.sh]: yeap

File: scripts/check_if_for_format.sh (hunk 1 above)
Comment (chriseth) [reply scripts/check_if_for_format.sh]: we own libdevcore, there is no upstream

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Let's merge this for now and add more rules as we see specific contributions that have incorrect style.

**Pull request merged**